commit 2123909652f67f3db445c7d057c38f57279b8a39
Author: Phil Elwell <phil@raspberrypi.org>
Date:   Mon May 8 16:30:18 2017 +0100

    ARM: dts: bcm283x: Reserve first page for firmware
    
    The Raspberry Pi startup stub files for multi-core BCM27XX processors
    make the secondary CPUs spin until the corresponding mailbox is
    written. These stubs are loaded at physical address 0x00000xxx (as seen
    by the ARMs), but this page will be reused by the kernel unless it is
    explicitly reserved, causing the waiting cores to execute random code.
    
    Use the /memreserve/ Device Tree directive to mark the first page as
    off-limits to the kernel.
    
    See: https://github.com/raspberrypi/linux/issues/1989
    
    Signed-off-by: Phil Elwell <phil@raspberrypi.org>

diff --git a/arch/arm/boot/dts/bcm2710-rpi-3-b.dts b/arch/arm/boot/dts/bcm2710-rpi-3-b.dts
index b21d2866d204..cbec919208f3 100644
--- a/arch/arm/boot/dts/bcm2710-rpi-3-b.dts
+++ b/arch/arm/boot/dts/bcm2710-rpi-3-b.dts
@@ -1,9 +1,5 @@
 /dts-v1/;
 
-#ifdef RPI364
-/memreserve/ 0x00000000 0x00001000;
-#endif
-
 #include "bcm2710.dtsi"
 #include "bcm283x-rpi-smsc9514.dtsi"
 
diff --git a/arch/arm/boot/dts/bcm283x.dtsi b/arch/arm/boot/dts/bcm283x.dtsi
index 7d58cd79ac6b..0bc1932cde43 100644
--- a/arch/arm/boot/dts/bcm283x.dtsi
+++ b/arch/arm/boot/dts/bcm283x.dtsi
@@ -3,6 +3,8 @@
 #include <dt-bindings/clock/bcm2835-aux.h>
 #include <dt-bindings/gpio/gpio.h>
 
+/memreserve/ 0x00000000 0x00001000;
+
 /* This include file covers the common peripherals and configuration between
  * bcm2835 and bcm2836 implementations, leaving the CPU configuration to
  * bcm2835.dtsi and bcm2836.dtsi.
